code/modules/power/solars/panel.dm

```diff
@@ -47,19 +64,77 @@
 			user.visible_message("<span class='notice'>[user] takes the [initial(G.name)] off the [src].</span>",\
 			"<span class='notice'>You take the [initial(G.name)] off the [src].</span>")
 			qdel(src)
+	if(iswrench(W) && !tracker)
+		var/target = input("Which orientation (in degrees) do you want the panel to face?","Set Orientation") as num|null
+		if (!isnull(target))
+			target = (360 + clamp(target,-360,360)) % 360//sanitizing
+			to_chat(user, "<span class='notice'>You begin rotating the solar panel.</span>")
+			playsound(src, 'sound/items/Ratchet.ogg', 50, 1)
+			spawn()
+				ndir = target
+				manual_rotation(user, target)
 	else if(W)
-		add_fingerprint(user)
+		user.do_attack_animation(src, user)
+		shake(1, 3)
+		playsound(src, 'sound/weapons/heavysmash.ogg', 75, 1)
+		add_hiddenprint(user)
 		user.delayNextAttack(10)
+		user.visible_message("<span class='warning'>[user] smashes \the [src] with \a [W].</span>", \
+							 "<span class='warning'>You smash \the [src] with \a [W].</span>")
 		health -= W.force
 		healthcheck()
 	..()
 
+/obj/machinery/power/solar/panel/proc/panel_rotation(var/target, var/speed)
+	if(adir != target)
+		var/direction = target - adir
```

Comment:
// Take the shortest rotation possible
var/direction = abs(direction) < 180 ? target - adir : adir - target